### PR TITLE
Add tests for User and Post models, and resolve n+1 problem

### PR DESCRIPTION
**Pull request by charlesgobina**
For this milestone, I
* Created tests for user and post models
* Added flash messages to create a post, like, and comments actions
* Solve n+1 problem in post controller

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      expect(post).to_not be_valid
+      post.comments_counter = 0
+    end
+
+    it 'like_counter should be integer' do
+      post.like_counter = 'string'
+      expect(post).to_not be_valid
+      post.like_counter = 0
+    end
+  end
```
Comment (wandji20) [inline spec/models/post_spec.rb:31]: - [ optional ] Nice work writing using tests for the post model. You may consider using `shoulda-matcher` to help achieve the same result in fewer lines.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -19,9 +19,9 @@ def create
     post = current_user.posts.new(post_params)
 
     if post.save
-      redirect_to user_posts_path(post.author)
+      redirect_to user_posts_path(post.author), notice: 'Post was successfully created.'
```
Comment (wandji20) [inline app/controllers/posts_controller.rb]: - [ ] Kindly use the `flash` object to successfully render all instances of flash messages in all controllers.

**Review by wandji20 — CHANGES_REQUESTED** [review]
Hello, @charlesgobina  :raised_hands: 
-----
Great :muscle:  job :raised_hands:  implementing most of the requirements for this project.
However, there are a few issues you need to address to have this project approved.
## To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Very descriptive pull request summary
:heavy_check_mark: Validations :100: 
:heavy_check_mark: N + 1 query solved :100: 

## Status: Require Changes :recycle: :red_circle: 
Please check the comments below and under the review and make all required changes.
Optional comments are not required to mark this project approved
## Required Changes
- [ ] We are still missing unit specs for the `Like` and `Comment` models. Kindly add them.
- [ ] Currently, your test and development database are both labeled ``blog`, Kindly rename them to match each environment(ie blog_development and blog_test for development and test respectively)

## Optional Changes
- N/A

**Happy coding :computer:** 
**_Do not hesitate to make comments in this PR in case of doubts._**
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb (hunk 1 above)
Comment (charlesgobina) [reply spec/models/post_spec.rb:31]: Hi @wandji20, thank you for the feedback. After renaming my database, will I need to create a new one and new migrations, as it won't correspond to the one on PostgreSQL

Comment (wandji20) [reply spec/models/post_spec.rb:31]: Exactly!

**Review by Gambit142 — APPROVED** [review]
Hi @charlesgobina 👋, 

Great work implementing the changes suggested by the other reviewer. 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by charlesgobina** [comment]
Thank you @Gambit142

**Pull request merged**